src/main/java/org/folio/rest/impl/CheckinReceivePiecesHelper.java

```diff
@@ -204,10 +206,15 @@ private PoLine searchPoLineById(List<PoLine> poLines, String polineId) {
     }
   }
 
+  private boolean ifLocationNotProcessed(String locationId) {
+    return processedLocations.add(locationId);
```

Comment:
The holding is defined as combination of `instanceId` + `locationId`. Instance id is linked to PO Line and the receiving API is designed to handle pieces related to different PO Lines which means - different instances